packages/app-shell/src/common/hooks/useUpdateSubscriptionPlan.js

```
@@ -22,12 +22,16 @@ const useUpdateSubscriptionPlan = ({
       awaitRefetchQueries: true,
       onCompleted: () => {
         ORGANIZATION_ACTIONS.billingUpdated({ user });
-      }
+      },
     }
   );
 
   useEffect(() => {
     if (!user || !plan || !hasPaymentMethod) {
+      // eslint-disable-next-line no-console
+      console.error(
```

I added this because when none of these things are present atm it silently fails and took a bit of time to find exactly what was causing the modals to get stuck in a processing state.

I didn't want to throw an error just yet as I haven't properly considered how that could affect current user flows and behaviors.

Do let me know if we would all prefer I just removed these new console errors I've added in this PR or if there's a preferred way 🙂

---

@daisymarie128 ideally we would want to use its dedicated mutation for the Quantity modal. 
[billingUpdateSubscriptionQuantity](https://github.com/bufferapp/core-authentication-service/blob/main/core-api/src/graphql/schema.graphql#L819-L824)
It is a simpler mutation on the call side, as it does not require the plan information (planId and interval). And also it is way simpler on the server side, as it has a single API call to Stripe instead of a multi-step logic that the update plan has. The reason is that only customers on a paid plan can purchase channels and that means that we don't have to handle their payment method, instantiate the Stripe form with an intend ID and proactivelly remove any active trials or cancelations (cancel at period end).

Long story short the above (new) mutation only increases the quantity of the subscription, which is actually what we need. 

That said we don't need to check for payment details either. Organizations with a paid plan, have already set that.